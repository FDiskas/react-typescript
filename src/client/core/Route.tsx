import * as React from 'react';
import { Route as ReactRoute, RouteProps } from 'react-router-dom';

export interface IRouteProps {
    layout?: React.ComponentType<RouteProps> | React.ComponentType<any>;
    component: React.ComponentType<RouteProps> | React.ComponentType<any>;
}

export interface IRouteState {
}

export class Route extends React.Component<IRouteProps & RouteProps, IRouteState> {
    constructor(props: IRouteProps) {
        super(props);
    }

    render() {
        const { component, layout, ...rest } = this.props;
        let routeComponent = (props: any) => React.createElement(component, props);

        if (layout) {
            routeComponent = props =>
                React.createElement(layout, props, React.createElement(component, props));
        }

        return <ReactRoute {...rest} render={routeComponent} />;
    }
}
