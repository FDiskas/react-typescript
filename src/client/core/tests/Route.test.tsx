import * as React from 'react';
import { shallow } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';

import { Route } from '../Route';

describe('Route.js', () => {
    const TestLayout = props => <div>{props}</div>;
    const TestComponent = () => <div>TestComponent</div>;

    it('renders without crashing', () => {
        shallow(<Route path={'/'} component={TestComponent} />);
    });

    it('matches snapshot', () => {
        const wrapper = shallow(<Route path={'/'} component={TestComponent} />);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it('matches snapshot', () => {
        const wrapper = shallow(<Route component={TestComponent} layout={TestLayout}/>);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });
});
