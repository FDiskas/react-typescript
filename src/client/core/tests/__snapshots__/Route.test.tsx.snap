// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route.js matches snapshot 1`] = `
<Route
  path="/"
  render={[Function]}
/>
`;

exports[`Route.js matches snapshot 2`] = `
<Route
  render={[Function]}
/>
`;
